kartothek/core/factory.py

```diff
@@ -99,20 +106,23 @@ def _instantiate_metadata_cache(self):
                 # backwards compat
                 self._cache_metadata = self._ds_callable()
             else:
-                self._cache_metadata = DatasetMetadata.load_from_store(
-                    uuid=self.dataset_uuid,
-                    store=self.store,
-                    load_schema=self.load_schema,
-                    load_all_indices=self.load_all_indices_flag,
-                )
+                self._exists = DatasetMetadata.exists(self.dataset_uuid, self.store)
+                if self._exists:
```

Comment:
What happens w/o this existence check?

Reply:
Simplified it. There is no need to check for existence here.